gapic/src/main/com/google/gapid/perfetto/models/Tracks.java

```diff
@@ -133,13 +137,29 @@ private Tracks() {
         .filter(c -> c.count > 0)
         .collect(toMap(c -> c.id, c -> c));
 
-    if (counters.isEmpty() && data.getGpu().isEmpty()) {
+    List<CounterInfo> gpuMemGlobalCounter = data.getCounters().values().stream()
```

Comment:
nit: it looks like an early filtering of `data.getCounters(CounterInfo.Type.Gpu)` can be helpful to avoid filtering too many counters.

Reply:
The GPU memory counters are separate from the GPU counters that come out from the vendor producer. The one that I added is a global counter and since its basically related to the GPU,  we are putting it inside the GPU group. This check is needed because even if the GPU counter producer is not present, this can output data and hence the GPU group should be present in the UI. Its a little repetitive because a stream cannot be operated more than once. The alternate I can think of is by iterating through the map and manually putting it to either the GPU counters or the global gpu memory usage.

Reply:
Sorry I pasted the wrong thing, I was meaning `data.getCounters(CounterInfo.Type. Global)`, which takes advantage of the existing map, and may ease some burden on java stream's filtering when there are too many counters.

Reply:
Discussed this offline. Closing this for now.